airflow/ti_deps/deps/trigger_rule_dep.py

```diff
@@ -379,7 +379,7 @@ def _evaluate_direct_relatives() -> Iterator[TIDepStatus]:
                     if skipped:
                         new_state = TaskInstanceState.SKIPPED
                 elif trigger_rule == TR.ALL_SKIPPED:
-                    if success or failed:
+                    if success or failed or upstream_failed:
                         new_state = TaskInstanceState.SKIPPED
```

Comment:
I think a better solution is to wait for a success or failed upstream to fast skip the TI, but if there is no upstream with this two states, all the upstream are done and at least one of them has upstream_failed status, we change the TI state to upstream_failed too:
```suggestion
                    if success or failed:
                        new_state = TaskInstanceState.SKIPPED
                    elif upstream_done and upstream_failed:
                        new_state = TaskInstanceState.UPSTREAM_FAILED
```
The reason for this suggestion is that we don't know the state of this upstream, maybe if when we run it, it will finish with skipped state.